src/test/unit/test_lumigo_utils.py

```diff
@@ -457,6 +456,13 @@ def test_get_edge_host(arg, host, monkeypatch):
 def test_report_json_extension_spans_mode(monkeypatch, reporter_mock):
     monkeypatch.setattr(Configuration, "should_report", True)
     monkeypatch.setenv("LUMIGO_USE_TRACER_EXTENSION", "TRUE")
+    mocked_urandom = MagicMock()
+    mocked_urandom.hex = MagicMock(return_value="my_mocked_data")
+    monkeypatch.setattr(os, "urandom", lambda *args, **kwargs: mocked_urandom)
+
+    start_span = [{"span": "true"}]
+    report_json(region=None, msgs=start_span, is_start_span=True)
```

Comment:
do you write files to the file system here?
This is highly not isolated. I.e. if you have these files and it stop working, then the next test will pass because the files are still there.
Please read: https://docs.pytest.org/en/6.2.x/tmpdir.html